File: 70-climbing-stairs/70-climbing-stairs.java
class Solution {
    
    HashMap<Integer,Integer> dp = new HashMap<Integer, Integer>();
    
    public int climbStairs(int n) {
        return climbStairsCount(n);    
    }
    
    public int climbStairsCount(int n){
        
        if(dp.containsKey(n)){
            return dp.get(n);
        }
        
        if(n==0){
            return 1;
        }
        if(n>=2){
            dp.put(n, climbStairsCount(n-1)+climbStairsCount(n-2));
            return dp.get(n);
        }else if(n==1){
            dp.put(n, climbStairsCount(n-1));
            return dp.get(n);
        }
        return 0;
    }
    
}

File: 70-climbing-stairs/70-climbing-stairs.py
class Solution(object):
    
    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """
        return self.differentClimbs(n,{})
    
    def differentClimbs(self,n, mapp):
        if n in mapp:
            return mapp[n]
        if n<0:
            return 0
        if n==0:
            return 1
        mapp[n-1] = self.differentClimbs(n-1, mapp)
        mapp[n-2] = self.differentClimbs(n-2, mapp)
        return mapp[n-1]+mapp[n-2]
            
        